packages/devtools_app/lib/src/flutter/device_dialog.dart

```diff
@@ -0,0 +1,265 @@
+  InfoController infoController;
+  TextEditingController filterController;
+
+  List<_DialogFlag> flags = [];
+  List<_DialogFlag> filteredFlags = [];
+
+  @override
+  void initState() {
+    super.initState();
+
+    infoController = InfoController();
+
+    filterController = TextEditingController();
+    filterController.addListener(() {
+      setState(() {
+        _refilter();
+      });
+    });
+
+    _updateFromController();
+    addAutoDisposeListener(infoController.flagListNotifier, () {
+      setState(() {
+        _updateFromController();
+      });
+    });
+  }
+
+  void _updateFromController() {
+    flags = infoController.flagListNotifier.value.flags
+        .map((flag) => _DialogFlag(flag))
+        .toList();
+    _refilter();
+  }
+
+  void _refilter() {
+    final filter = filterController.text.trim().toLowerCase();
+
+    if (filter.isEmpty) {
+      filteredFlags = flags;
+    } else {
+      filteredFlags =
+          flags.where((flag) => flag.filterText.contains(filter)).toList();
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final textTheme = Theme.of(context).textTheme;
+
+    return AlertDialog(
+      actions: [
+        DialogCloseButton(),
+      ],
+      content: Column(
+        mainAxisSize: MainAxisSize.min,
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
+          Row(
+            children: [
+              Text('VM Flags', style: textTheme.headline6),
+              const Expanded(child: SizedBox(width: denseSpacing)),
+              Container(
+                width: 200.0,
+                height: 36.0,
+                child: TextField(
+                  controller: filterController,
+                  decoration: const InputDecoration(
+                    isDense: true,
+                    border: OutlineInputBorder(),
+                    labelText: 'Filter',
+                  ),
+                ),
+              ),
+            ],
+          ),
+          const PaddedDivider(
+              padding: EdgeInsets.only(bottom: denseRowSpacing)),
+          SizedBox(
+            width: 875,
+            height: 375,
+            child: _FlagTable(filteredFlags),
+          ),
+        ],
+      ),
+    );
+  }
+
+  @override
+  void dispose() {
+    infoController.dispose();
+
+    super.dispose();
+  }
+}
+
+class _FlagTable extends StatelessWidget {
+  _FlagTable(this.flags);
+
+  final List<_DialogFlag> flags;
+
+  final ColumnData<_DialogFlag> name = _NameColumn();
+  final ColumnData<_DialogFlag> description = _DescriptionColumn();
+  final ColumnData<_DialogFlag> value = _ValueColumn();
+
+  List<ColumnData<_DialogFlag>> get columns => [name, description, value];
+
+  @override
+  Widget build(BuildContext context) {
+    return OutlineDecoration(
+      child: FlatTable<_DialogFlag>(
+        columns: columns,
+        data: flags,
+        autoScrollContent: true,
```

Comment:
I'm surprised that we would be autoscrolling content to the end. Why do we need to do this?

Reply:
We definitely don't good catch. This was a copy paste error.